provider/blob/src/blob_data_provider.rs

```diff
@@ -150,61 +150,90 @@ mod test {
 
     #[test]
     fn test_empty() {
-        let mut blob: Vec<u8> = Vec::new();
+        let mut version = 1;
 
-        {
-            let mut exporter = BlobExporter::new_with_sink(Box::new(&mut blob));
+        while version <= 2 {
+            let mut blob: Vec<u8> = Vec::new();
 
-            exporter.flush(HelloWorldV1Marker::KEY).unwrap();
+            {
```

Comment:
why this block?

Reply:
It was there before, I think because BlobExporter holds a mutable borrow on the Vec which we need to release once it is flushed

Reply:
oh yeah it was